src/sources/aws_s3/sqs.rs

```diff
@@ -614,6 +614,10 @@ impl IngestorProcess {
         // so we explicitly drop it so that we can again utilize `read_error` below.
         drop(stream);
 
+        // The BatchNotifier is cloned for each LogEvent in the batch stream, but the last
+        // reference must be dropped before the status of the batch is sent to the channel.
+        drop(batch);
```

Comment:
This should actually be done before the send, as the events themselves hold a copy of the batch during the send.

Reply:
Thanks @bruceg, good catch - fixed in 45dc94f

Reply:
Thanks @bruceg, good catch - fixed in 45dc94f